inc/az_ulib_ustream_forward_base.h

```diff
@@ -0,0 +1,374 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+#ifndef AZ_ULIB_ustream_forward_BASE_H
+#define AZ_ULIB_ustream_forward_BASE_H
+
+#include "az_ulib_config.h"
+#include "az_ulib_pal_os.h"
+#include "az_ulib_result.h"
+#include "azure/core/az_span.h"
+#include "azure/core/az_context.h"
+
+#ifdef __cplusplus
+#include <cstddef>
+#include <cstdint>
+#else
+#include <stddef.h>
+#include <stdint.h>
+#endif /* __cplusplus */
+
+#include "azure/core/_az_cfg_prefix.h"
+
+/**
+ * @brief   Define offset_t with the same size as size_t.
+ */
+typedef size_t offset_t;
+
+/**
+ * @brief   Forward declaration of az_ulib_ustream_forward. See #az_ulib_ustream_forward_tag for struct members.
+ */
+typedef struct az_ulib_ustream_forward_tag az_ulib_ustream_forward;
+
+/**
+ * @brief Signature of the function to be invoked by the `ustream_forward->flush` operation when the 
+ *        `const az_span* const` buffer has been created.
+ * 
+ * @param[in]   buffer                  The `const az_span* const` buffer to be handled by the 
+ *                                      implementation of this callback.
+ * @param[in]   push_callback_context   The #az_context* contract held between the owner of this
```

Comment:
>az_context [](http://example.com/codeflow?start=45&length=10)

Looks like this az_context is specfific for az_core. Consider create a simple one for this stream. It can be just a `void*`.